main.tf

```diff
@@ -85,7 +96,9 @@ locals {
 
   labels = [for l in local.label_order : local.id_context[l] if length(local.id_context[l]) > 0]
 
-  id_full = lower(join(local.delimiter, local.labels))
+  id_full = local.id_case == "lower" ? lower(join(local.delimiter, local.labels)) : (
+    local.id_case == "title" ? title(join(local.delimiter, local.labels)) :
+  upper(join(local.delimiter, local.labels)))
```

Comment:
As with `generated_tags`, use a map to validate the inputs. Include `none` (no case modification) as an option.